### DISCUSSION TITLE
Including in Debian?

### DISCUSSION THREAD

**Discussion by philippemilink** [dionysius/vaultwarden-deb#10] [General]
Hello,

Is there a reason not to send this package directly to Debian? There is an [ITP (*Intent to Package*) bug about Vaultwarden](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1067023). 

I would very much appreciate to have Vaultwarden available in Debian, let me know if I can help to make it happen.

**Comment by dionysius** [comment]
Hey, ~~from a technical perspective none~~, some workaround stuff needs to be removed and built against unstable. But from a organisational perspective AFAIK there are some strings attached: A debian package maintainer should be able to maintain a package long term, which I'm unsure I can provide. I'm happy to help to bring this into a for debian acceptable state, but probably someone else should be willing to maintain it. I intentionally licenced the debian files as expat to hopefully make it easy for someone to pickup.

**Reply by dionysius** [reply]
Correction about the technical part: There will be some work needed to build it cleanly with dh-cargo (https://github.com/dionysius/vaultwarden-deb/blob/main/debian/rules#L24). As well in debian any dependency must already been built and available in debian and those must be used in the build process. I'm no rust dev so maintaining something I'm not familiar is also a point for me not be able to maintain this package officially.

**Comment by guerzon** [comment]
Hello, I have been working on packaging the missing Rust crates that Vaultwarden depends on. Some of them are below:
- openidconnect
- yubico_ng
- cron and job_scheduler_ng (packaging exists but have to be uploaded)
- devise_core
- stable-pattern
Indeed, these will eventually have to be maintained. Even if we could somehow complete all the dependencies and package Vaultwarden, it would require maintenance in the future. We can't just push and later leave the Debian Rust team to do all the work.
So I would like to ask: is there anybody who genuinely has a requirement to push Vaultwarden to the Debian archives?

**Reply by dionysius** [reply]
> So I would like to ask: is there anybody who genuinely has a requirement to push Vaultwarden to the Debian archives?

**Everybody is invited. We'd like to hear your preferences**

---

Thank you @guerzon for your work! I host myself a vaultwarden for family and friends, each using whatever client they just happen to have (probably pretty latest due to app stores). I would like to use a debian maintained vaultwarden, but we have to account for api changes with newer bitwarden clients - we unfortunately cannot pin clients. Do you see backports as a possible solution for providing more recent versions of vaultwarden? Otherwise I see vaultwarden in debian only useful for web-ui-only users (and maybe third-party clients - if any exist)

<details>
<summary>AI generated list of client issues last 2 years (related to contract or api changes)</summary>

| Fix date | VW release | What broke for users |
|---|---|---|
| 2024-09-09 | 1.32.1 | Sync fails with the new native mobile apps ([dca1428](https://github.com/dani-garcia/vaultwarden/commit/dca1428)) |
| 2024-10-13 | 1.32.3 | **iOS: "An error has occurred", vault won't sync** — custom-field `type` sent as string ([#5069](https://github.com/dani-garcia/vaultwarden/issues/5069), [49c5dec](https://github.com/dani-garcia/vaultwarden/commit/49c5dec)) |
| 2024-12-08 | 1.32.6 | Sync fails on `reprompt=4` — native clients validate strictly ([c9860af](https://github.com/dani-garcia/vaultwarden/commit/c9860af)) |
| 2025-01-23 | 1.33.0 | `/api/devices` missing; pending device-login requests dead ([c0be36a](https://github.com/dani-garcia/vaultwarden/commit/c0be36a)) |
| 2025-01-28 | 1.33.1 | Icons broken in desktop clients ([1b46c80](https://github.com/dani-garcia/vaultwarden/commit/1b46c80)) |
| 2025-07-01 | 1.34.2 | **Cannot delete or restore items** — clients ≥2025.6.0 read a `permissions` object the server never sent ([3b48e6e](https://github.com/dani-garcia/vaultwarden/commit/3b48e6e)) |
| 2025-08-09 | 1.35.0 | **Email 2FA login broken on mobile** — app switched to `DeviceIdentifier` (PascalCase) ([4a5516e](https://github.com/dani-garcia/vaultwarden/commit/4a5516e)) |
| 2025-12-19 | 1.35.0 | `accountKeys` / `masterPasswordUnlock` required in the identity response ([061d320](https://github.com/dani-garcia/vaultwarden/commit/061d320)) |
| 2026-03-09 | 1.35.5 | **New desktop app can't connect at all** — new CORS origin `bw-desktop-file://bundle` ([1a1d7f5](https://github.com/dani-garcia/vaultwarden/commit/1a1d7f5)) |
| 2026-04-28 | 1.36.0 | **Fresh logins fail entirely** — `/identity/accounts/prelogin/password` didn't exist ([fd2b652](https://github.com/dani-garcia/vaultwarden/commit/fd2b652)) |
| 2026-07-24 | 1.37.0 | New item types (bank account, driver's licence, passport) ([a6a88e7](https://github.com/dani-garcia/vaultwarden/commit/a6a88e7)) |

| `bitwarden/server` response model | changes 2024-08 → 2026-08 |
|---|---|
| `SyncResponseModel` | 18 |
| `OrganizationUserResponseModel` | 16 |
| `CipherResponseModel` | 13 |
| `ConfigResponseModel` | 10 |

</details>

Update: and just now dropped [1.37.2](https://github.com/dani-garcia/vaultwarden/releases/tag/1.37.2), with:
> This update is required for support with clients with version 2026.8.0+, please update before reporting any issues with them.

**Reply by guerzon** [reply]
Hi @dionysius, I mentally responded to this post but apparently I forgot to actually respond IRL :-)

I haven't taken into account Bitwarden app clients as I don't use them (I only use the web). But this is indeed a showstopper.

My Debian packaging experience is limited (since I only started contributing last month), but if my understanding is correct, backports (which tracks unstable) could also be a lot of work since we will have to maintain baseline dependencies and make sure nothing breaks. From https://backports.debian.org/Contribute/:

> Please note, that you are responsible for this backport from the time on when it was accepted on debian-backports. This means, you have to keep track of the changes in unstable, update your backport when a new version enters testing and provide security updates when needed. If you are not willing or capable of doing this, you better ask someone else (e.g. on the [mailing list](https://lists.debian.org/debian-backports/)) to create and maintain the backport.

Looking for a DD to do the upload for every change (not only for vaultwarden in backports but also for any baseline dependencies that were backported) also presents another challenge.

**Reply by dionysius** [reply]
I took my time to think about it. First of all the reach of this discussion post is probably close to zero. Worth considering moving this to the ITP.

I have the following thoughts and suggestions: I believe debian would benefit from a official vaultarden package. But I don't think the effort is worth to constantly update the source and dependencies to bring in as backports - the user base is not there (yet) - it is not a browser where millions use it. And to get a user base it needs to start somewhere. If possible, declare clearly that there's no support for clients other than the provided web vault. Security patches should still be applied. For users using other clients, the middle line might be a project like this here - they can choose anyway which source fits their requirements best. (btw. the apt from this is also listed in extrepo).

**Reply by guerzon** [reply]
Sounds good to me. I will continue working on packaging the dependencies.
